### PR TITLE
Fix some bugs in order to run in the big data

### PR DESCRIPTION
**Pull request by anbangx**
@JavierJia @jakebiesinger @Elmira88 @Nan-Zhang 

When we try to run in the big data, there are some bugs. This PR fixed them. Have a look!

### CODE REVIEW

**Comment by jakebiesinger** [comment]
It seems the only differences are to set the max memory for the stats job
like I was saying and to set the read id set to actually dump to disk (I
think this patch is already in one of the branches?)

Or was the problem in generating the histogram for the sorted counter min
length?

File: genomix/genomix-data/src/main/java/edu/uci/ics/genomix/data/types/ExternalableTreeSet.java [hunk 1]
```diff
@@ -29,7 +29,8 @@
      */
     private static final long serialVersionUID = 1L;
     static FileManager manager;
-    static private int countLimit = Integer.MAX_VALUE;
+//    static private int countLimit = Integer.MAX_VALUE;
+    static private int countLimit = 1000;
```
Comment (JavierJia) [inline genomix/genomix-data/src/main/java/edu/uci/ics/genomix/data/types/ExternalableTreeSet.java:33]: we should remove this debug line. and recover the original commented out line.

File: genomix/genomix-data/src/main/java/edu/uci/ics/genomix/data/types/ExternalableTreeSet.java [hunk 2]
```diff
@@ -253,6 +254,7 @@ public void write(DataOutput out) throws IOException {
             } else if (isChanged) {
                 save(path, inMemorySet);
             }
+            System.err.println("Write to HDFS:" + path);
```
Comment (JavierJia) [inline genomix/genomix-data/src/main/java/edu/uci/ics/genomix/data/types/ExternalableTreeSet.java:257]: remove this debug message.

File: genomix/genomix-driver/src/main/java/edu/uci/ics/genomix/driver/GenomixDriver.java [hunk 3]
```diff
@@ -198,8 +198,8 @@ private void addStep(GenomixJobConf conf, Patterns step) throws Exception {
             case MERGE:
             case MERGE_P4:
                 pregelixJobs.add(P4ForPathMergeVertex.getConfiguredJob(conf, P4ForPathMergeVertex.class));
-                flushPendingJobs(conf);
-                setMinScaffoldingSeedLength(conf);
+               // flushPendingJobs(conf);
+               // setMinScaffoldingSeedLength(conf);
```
Comment (JavierJia) [inline genomix/genomix-driver/src/main/java/edu/uci/ics/genomix/driver/GenomixDriver.java:202]: Recover these two lines.

File: genomix/genomix-hadoop/src/main/java/edu/uci/ics/genomix/hadoop/utils/GraphStatistics.java [hunk 4]
```diff
@@ -73,6 +80,7 @@ public void map(VKmer key, Node value, OutputCollector<Text, LongWritable> outpu
             throws IOException {
 
         this.reporter = reporter;
+        System.err.println("Inside the Map");
```
Comment (JavierJia) [inline genomix/genomix-hadoop/src/main/java/edu/uci/ics/genomix/hadoop/utils/GraphStatistics.java:83]: remove this debug line

File: genomix/genomix-hadoop/src/main/java/edu/uci/ics/genomix/hadoop/utils/GraphStatistics.java [hunk 5]
```diff
@@ -130,6 +138,7 @@ public static Counters run(String inputPath, String outputPath, JobConf baseConf
         JobConf conf = new JobConf(baseConf);
         conf.setJobName("Graph Statistics");
         conf.setMapperClass(GraphStatistics.class);
+        conf.set("mapred.child.java.opts", "-Xmx50g");
```
Comment (JavierJia) [inline genomix/genomix-hadoop/src/main/java/edu/uci/ics/genomix/hadoop/utils/GraphStatistics.java:141]: remove this debug line.

**Comment by JavierJia** [comment]
@jakebiesinger , the problem was to set the count limit and initialize the manager before each job. 
Otherwise the default countLimit of `ReadHeadSet` is `Interger.Max`. Then every readFields method will load from the frame instead of the hdfs files. 

We call the `setGlobalStaticConstants` when the job initializing.

**Comment by JavierJia** [comment]
Hi, @anbangx , I just saw this branch merged, but those comments didn't cover ?

**Comment by anbangx** [comment]
@JavierJia I changed them based on your comments already. We tried to run as soon as possible, so I didn't send it again. Except this: 
+//    static private int countLimit = Integer.MAX_
-    static private int countLimit = 1000;

**Pull request merged**